lib/Service/ContextChat/TaskService.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+namespace OCA\Mail\Service\ContextChat;
+
+use OCA\Mail\Db\ContextChat\Task;
+use OCA\Mail\Db\ContextChat\TaskMapper;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\MultipleObjectsReturnedException;
+use OCP\DB\Exception;
+
+class TaskService {
+	public function __construct(
+		private TaskMapper $taskMapper,
+	) {
+	}
+
+	/**
+	 * @return Task
+	 * @throws DoesNotExistException
+	 * @throws Exception
+	 * @throws MultipleObjectsReturnedException
+	 */
+	public function findNext(): Task {
+		return $this->taskMapper->findNext();
+	}
+
+	/**
+	 * Update job, or create it if it doesn't exist
+	 *
+	 * @param int $mailboxId
+	 * @param int $lastMessageId
+	 * @return Task
+	 * @throws Exception|\OCP\AppFramework\Db\MultipleObjectsReturnedException
+	 */
+	public function updateOrCreate(int $mailboxId, int $lastMessageId): Task {
+		try {
+			$entity = $this->taskMapper->findByMailbox($mailboxId);
+		} catch (DoesNotExistException) {
+			$entity = new Task();
+			$entity->setMailboxId($mailboxId);
+			$entity->setLastMessageId($lastMessageId);
+
+			return $this->taskMapper->insert($entity);
+		}
+
+		if ($lastMessageId >= $entity->getLastMessageId()) {
+			// Existing job already starts at an earlier message, so updating the database is not needed
+			return $entity;
+		}
+
+		$entity->setLastMessageId($lastMessageId);
```

Comment:
Something appears to be wrong here. `last_message_id` is initialized with `0`. It will never be updated

<img width="1337" height="654" alt="Bildschirmfoto vom 2025-11-07 14-19-16" src="https://github.com/user-attachments/assets/270521b5-4e2c-4544-bc0d-dee3a433c4b6" />
<img width="631" height="654" alt="image" src="https://github.com/user-attachments/assets/c2351aa5-7941-495c-b28a-236e1d564bcc" />

and that will lead to the same message being queued again and again

<img width="1414" height="403" alt="Bildschirmfoto vom 2025-11-07 14-17-03" src="https://github.com/user-attachments/assets/d0b11ece-2051-40df-b0f6-75bcec8d51de" />